utils/bytes.go

```diff
@@ -14,3 +21,62 @@ func CopyBytes(b []byte) []byte {
 	copy(cb, b)
 	return cb
 }
+
+// MarshalBytes returns []byte(v) if v is a string
+// or marshals v as JSON bytes.
+func MarshalBytes(v interface{}) ([]byte, error) {
+	switch value := v.(type) {
+	case string:
+		return []byte(value), nil
+	default:
+		b, err := json.Marshal(value)
+		if err != nil {
+			return nil, fmt.Errorf("could not marshal JSON: %w", err)
+		}
+		return b, nil
+	}
+}
+
+// SetByteSlices calls MarshalBytes on
+// the value of all keys found in source that are
+// also a field in v, setting the value of the field
+// to the result.
+//
+// Note: Field names are lowercased before looking
+// up corresponding keys in source.
+//
+// Note: v MUST be a pointer for this function
+// to work correctly.
+func SetByteSlices(source map[string]interface{}, v interface{}) error {
+	s := reflect.ValueOf(v).Elem()
+	for i := 0; i < s.NumField(); i++ {
+		field := s.Field(i)
+		fieldType := s.Type().Field(i)
+		if fieldType.Type.Kind() != reflect.Slice {
+			continue
+		}
+
+		if fieldType.Type.Elem().Kind() != reflect.Uint8 {
+			continue
+		}
+
+		if !field.CanSet() {
+			continue
+		}
+
+		name := strings.ToLower(fieldType.Name)
```

Comment:
I think it might be better not to enforce that everything is lower case

Reply:
This is actually required otherwise we will never populate any fields (thanks to `viper`). Let me explain....

If the struct field is capitalized, that means it is public and we can set it. However, the key in the `source` map will always be lowercase. To match the field in `source` to the struct field, we must lowercase the struct field.

If the struct field is lowercase, that means it isn't public and we can't set it. Even if we have a match in `source`, we can't do anything with it.